docs/readmes/lte/GTPU_Extension_Header.md

```diff
@@ -0,0 +1,139 @@
+
+### Transfer of PDU Session Information for Downlink Data Traffic
+
+The Transfer of PDU Session Information for downlink data packets involves transfer of control information elements related to the PDU Session from Pipelined to NG-RAN.
+
+The DL PDU SESSION INFORMATION frame includes a QoS Flow Identifier (QFI) field associated with the transferred packet. The NG-RAN uses the received QFI to determine the QoS flow and QoS profile which are associated with the received packet.
+
+The DL PDU session information frame includes the Reflective QoS Indicator (RQI) field to indicate whether the user plane reflective QoS is to be activated or not. This is only applicable if reflective QoS is activated.
+
+## High Level Design
+
+The following functionality is supported for QFI set in GTP Extension header:
+QFI Configuration for uplink flow (Matching criteria)
+QFI Configuration for downlink flow (Action criteria)
+Uplink traffic match with uplink flow
+Downlink traffic takes action with action criteria values.
+
+### Pipelined
+
+Pipelined will extract qos structure from uplink PDR which is coming from SMF. QFI value gets from QoS structure and sets it in uplink flow as a match criteria.
+                       match = MagmaMatch(tunnel_id=i_teid, qfi=<qfi_value>, in_port=gtp_portno)
+Pipelined will extract qos structure from downlink PDR which is coming from SMF. QFI value gets from QoS structure and sets it in downlink flow as an action criteria.
+                       actions.append(parser.OFPActionSetField(qfi = <qfi_value>)
+
+### RYU Controller
+
+Ryu is a Python library that provides an API wrapper for programming OVS.
+
+There are two parts to taking care of QFI functionality support:
+Nicira Extension Actions Structures (NXAction)
+Nicira Extended Match Structures (OFPMatch)
+
+### OVS Openflow
+
+Using the OpenFlow protocol, the controller can add, update, and delete flow entries in flow tables, both reactively (in response to packets) and proactively. Each flow table in the switch contains a set of flow entries; each flow entry consists of match fields, counters, and a set of instructions to apply to matching packets.
+
+### OVS Kernel
+
+For Egress flow, one needs to build the extension header and set the QFI value. It needs to be Passed down to kernel datapath for pushing it to GTP-U packets.
+For Ingress flow, extract the QFI value from skb of GTP header and set to the tunnel key.
+
+## High Level Call Flow
+
+Call flow for QFI set in OVS.
+
+```mermaid
+sequenceDiagram
+    CPE->>MME(AMF): PDU Session Establishment Request
+    MME(AMF)->>SESSIOND: SetSMFContext
+    SESSIOND->>PIPELINED: SetSMFSession(GTP)
+    PIPELINED->>RYU_CONTROLLER: Set QFI value in match and action criteria
+    RYU_CONTROLLER->>OPENFLOW: NXM_NX_QFI
+    OPENFLOW->>OVS_DATAPATH: MFF_QFI and set tunnel QFI
+    CPE->>OVS_DATAPATH: GTP Traffic for uplink/downlink with QFI value
+```
+
+## Common Issues and Troubleshooting
+

```

Comment:
These are generic debugging steps can you add it to datapath debugging doc?
https://docs.magmacore.org/docs/howtos/troubleshooting/datapath_connectivity

Reply:
Yes, few generic debugging steps are moved to datapath_connectivity.md